### PR TITLE
MBS-10910: Display renamed from/into on label overview

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-10910

This seems like a sensible way to follow the flow of labels that were renamed, and I see no negatives about showing them there.
I saw artist legal name is treated as only ever one result (which seems weird to me, but that's a separate issue). Because a label could be listed as renamed from or into multiple others, I pass these as arrays. I expect in most cases these would be data errors, but having them be visible makes them more likely to get fixed anyway.

### CODE REVIEW

**Comment by reosarevok** [comment]
> It shows the latest previous name and the first next name only 

IMO that's correct - only those were renamed to and from this label. Any others were renamed to and from the other labels.

**Comment by yvanzo** [comment]
This relationship seems to be transitive, so it seems it would be correct to show all the other names too. In either case, dates apply.

**Comment by reosarevok** [comment]
Wouldn't the dates be just a duplicate of the label start and end dates?

File: lib/MusicBrainz/Server/Controller/Label.pm [hunk 1]
```diff
@@ -117,11 +119,29 @@ sub show : PathPart('') Chained('load')
     $c->model('MediumFormat')->load(map { $_->all_mediums } @$releases);
     $c->model('ReleaseLabel')->load(@$releases);
  
+    my @renamed_from = map { $_->target }
+                        grep { $_->direction == $MusicBrainz::Server::Entity::Relationship::DIRECTION_BACKWARD }
+                        grep { $_->link->type->gid eq 'e6159066-6013-4d09-a2f8-bc473f21e89e' } @{ $label->relationships };
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Controller/Label.pm:124]: suggesting to store the first grep in a var, then use it for both `@renamed_from` and `@renamed_to`.
```suggestion
    my @rename_rels = grep { $_->link->type->gid eq 'e6159066-6013-4d09-a2f8-bc473f21e89e' } @{ $label->relationships };
    my @renamed_from = map { $_->target }
                        grep { $_->direction == $MusicBrainz::Server::Entity::Relationship::DIRECTION_BACKWARD }
                        @rename_rels;
```

**Review by mwiencek — APPROVED** [review]

File: root/label/LabelIndex.js [hunk 2]
```diff
@@ -51,6 +59,20 @@ const LabelIndex = ({
...
     />
+    {renamedFrom.length ? (
+      <RelatedEntitiesDisplay title={l('Renamed from')}>
+        {commaOnlyList(renamedFrom.map(
+          label => <DescriptiveLink entity={label} key={label.gid} />,
+        ))}
+      </RelatedEntitiesDisplay>
+    ) : null}
+    {renamedInto.length ? (
+      <RelatedEntitiesDisplay title={l('Renamed into')}>
```
Comment (mwiencek) [inline root/label/LabelIndex.js:70]: This doesn't seem wrong but "Renamed to" sounds a bit more common.
```suggestion
      <RelatedEntitiesDisplay title={l('Renamed to')}>
```

File: root/label/LabelIndex.js [hunk 3]
```diff
@@ -51,6 +59,20 @@ const LabelIndex = ({
       entity={label}
       numberOfRevisions={numberOfRevisions}
     />
+    {renamedFrom.length ? (
+      <RelatedEntitiesDisplay title={l('Renamed from')}>
```
Comment (mwiencek) [inline root/label/LabelIndex.js:63]: This one sounds slightly strange to me though without both parts, as in "renamed from x to y." I'd suggest "Previously named" or "Previously known as." Do you think either of those would work?

```suggestion
      <RelatedEntitiesDisplay title={l('Previously named')}>
```

**Review by mwiencek — COMMENTED** [review]
I rebased and pushed some minor changes here to use only one `load_subset` call -- not that it's common for both directions to exist, but might as well.

But looking at it again, I'm wondering if we can improve the wording slightly.  Let me know what you think.

File: root/label/LabelIndex.js (hunk 3 above)
Comment (reosarevok) [reply root/label/LabelIndex.js:63]: I like "Previously known as" a bit more so using that.

**Comment by mwiencek** [comment]
Thanks!  There hasn't been any follow-up about the dates in almost a year so I think it's best to merge this, as it's an improvement; it can certainly be improved further later on if people ask for dates.

**Pull request merged**